src/DataAccess/ApiBasedPageRetriever.php

```diff
@@ -28,14 +28,31 @@ public function __construct( MediawikiApi $api, ApiUser $apiUser, LoggerInterfac
 		$this->logger = $logger;
 	}
 
-	public function fetchPage( string $pageTitle ): string {
+	/**
+	 * @param string $pageTitle
+	 * @param string $action
+	 * @throws \RuntimeException if the value of $action is not supported
+	 * @return string
+	 */
+	public function fetchPage( string $pageTitle, string $action = 'render' ): string {
 		$this->logger->debug( __METHOD__ . ': pageTitle', [ $pageTitle ] );
 
 		if ( !$this->api->isLoggedin() ) {
 			$this->doLogin();
 		}
 
-		$content = $this->retrieveRenderedPage( $pageTitle );
+		$content = false;
```

Comment:
Looks like this assignment has no effect

Perhaps its worthwhile to factor out the switch into a `retrieveContent` method?